File: X-Clone-Backend-FastAPI/app/models/retweet.py
from __future__ import annotations

from sqlalchemy import Column, Integer, String, TIMESTAMP

from ..db import Base


class Retweet(Base):
    __tablename__ = "retweets"

    id = Column(Integer, primary_key=True, autoincrement=True)
    reference_id = Column(Integer, nullable=False)
    retweeter_id = Column(Integer, nullable=False)
    type = Column(String(32), nullable=False)
    created_at = Column(TIMESTAMP, nullable=True)

File: X-Clone-Backend/src/main/java/com/xclone/xclone/domain/retweet/Retweet.java
package com.xclone.xclone.domain.retweet;

import jakarta.persistence.*;

import java.sql.Timestamp;

@Entity
@Table (name = "retweets")
public class Retweet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "reference_Id")
    private Integer referenceId;

    @Column(name = "retweeter_id")
    private Integer retweeterId;

    @Column(name = "type")
    private String type;

    @Column(name = "created_at", updatable = false, insertable = false)
    private Timestamp createdAt;

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getReferenceId() {
        return referenceId;
    }

    public void setReferenceId(Integer referenceId) {
        this.referenceId = referenceId;
    }

    public Integer getRetweeterId() {
        return retweeterId;
    }

    public void setRetweeterId(Integer retweeterId) {
        this.retweeterId = retweeterId;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

}
